src/sentry/seer/anomaly_detection/store_data.py

```diff
@@ -49,7 +48,12 @@ def format_historical_data(data: SnubaTSResult) -> list[TimeSeriesPoint]:
     return formatted_data
 
 
-def send_historical_data_to_seer(alert_rule: AlertRule, project: Project) -> BaseHTTPResponse:
+def send_historical_data_to_seer(
```

Comment:
I had to refactor this to handle the case where we're updating an alert but the Seer call fails - if I let the rule update and _then_ did the Snuba query and Seer call, we risk having an alert in a bad state. I decided to pass all the data needed to the function instead - for edited alerts we already have the rule ID so that's fine, and the rest I just put into new typeddicts.

Reply:
Hmm...

If the concern here is that the alert_rule could be saved in a bad state, rather than using `alert_rule.update`, we could just set the attrs on the `alert_rule` and just not `alert_rule.save()` it until we've properly updated seer?

> having an alert in a bad state
could you explain more of what state the alert rule could fall into?

Reply:
The bad state would be that we've already updated it but the Seer call fails so we now have a dynamic alert rule with a metric that has no historical data, or outdated data for a different metric

Reply:
I changed this to loop through and set the attrs for dynamic alerts only before we call `send_historical_data_to_seer` and then `update()`